app/src/main/java/app/musikus/ui/activesession/ActiveSessionUiState.kt

```diff
@@ -8,20 +8,144 @@
 
 package app.musikus.ui.activesession
 
-import app.musikus.database.Nullable
-import app.musikus.database.daos.LibraryFolder
+import androidx.compose.runtime.Stable
+import androidx.compose.ui.graphics.Color
+import app.musikus.database.LibraryFolderWithItems
+import app.musikus.database.UUIDConverter
 import app.musikus.database.daos.LibraryItem
-import app.musikus.ui.library.DialogMode
-import app.musikus.ui.library.LibraryItemDialogUiEvent
-import app.musikus.ui.library.LibraryItemDialogUiState
-import app.musikus.ui.library.LibraryItemEditData
+import kotlinx.coroutines.flow.MutableStateFlow
+import kotlinx.coroutines.flow.StateFlow
 import java.time.ZonedDateTime
 import java.util.UUID
-import kotlin.time.Duration
 
 
+// TODO UIText for all Strings?
 
+@Stable
+enum class SessionPausedResumedState {
+    RUNNING,
+    PAUSED
+}
+
+/** Specific one-time events triggering some UI action in ActiveSession */
+@Stable
+data class ActiveSessionEventStates (
+    val sessionSaved: Boolean = false,
+    val sessionDiscarded: Boolean = false,
+)
+
+@Stable
+/** Main UI State */
 data class ActiveSessionUiState(
+    val topBarUiState: StateFlow<ActiveSessionTopBarUiState> = MutableStateFlow(ActiveSessionTopBarUiState()),
+    val mainContentUiState: StateFlow<MainContentUiState> = MutableStateFlow(MainContentUiState()),
+    val newItemSelectorUiState: StateFlow<NewItemSelectorUiState> = MutableStateFlow(NewItemSelectorUiState()),
+    val toolsUiState: StateFlow<ActiveSessionToolsUiState> = MutableStateFlow(ActiveSessionToolsUiState()),
+    val dialogVisibilities: StateFlow<DialogVisibilities> = MutableStateFlow(DialogVisibilities()),
+)
+
+data class DialogVisibilities(
+    val newItemSelectorVisible: Boolean = false,
+    val finishDialogVisible: Boolean = false,
+    val discardDialogVisible: Boolean = false,
+    val createItemDialogVisible: Boolean = false,
+    val createFolderDialogVisible: Boolean = false,
+)
+
+@Stable
+data class MainContentUiState(
+    val timerUiState: StateFlow<ActiveSessionTimerUiState> = MutableStateFlow(ActiveSessionTimerUiState()),
+    val currentItemUiState: StateFlow<ActiveSessionCurrentItemUiState> = MutableStateFlow(ActiveSessionCurrentItemUiState()),
+    val pastSectionsUiState: StateFlow<ActiveSessionCompletedSectionsUiState> = MutableStateFlow(ActiveSessionCompletedSectionsUiState()),
+    val endDialogUiState: StateFlow<ActiveSessionEndDialogUiState> = MutableStateFlow(ActiveSessionEndDialogUiState())
+)
+
+@Stable
+data class ActiveSessionTopBarUiState(
+    val visible: Boolean = false,
+    val pauseButtonAppearance: SessionPausedResumedState = SessionPausedResumedState.RUNNING,
+)
+
+@Stable
+data class ActiveSessionTimerUiState(
+    val timerText: String = "00:00", // Prevents size change animation to trigger when opening the screen
+    val subHeadingAppearance: SessionPausedResumedState = SessionPausedResumedState.RUNNING,
```

Comment:
Imo the type and name of a property should match better. I'd call the variable `isSessionPaused` and make it a Boolean.

Reply:
I wanted to define the UI state as "close" as possible to the actual description of *how it looks* not *what type of data it gets and interprets it into a specific appearance*. It's a minor thing, but I tried to define UIState variables as much as possible of a description of "how it looks" and not *why it looks as it looks*.

It's a question of taste I guess, also not to say that `subHeadingAppearance` is an especially well name for it.

Reply:
Well in this specific case, the property name refers to _how it looks_ but the type still refers to _why it looks the way it does_. That's the inconsistency which irritates me I think 🙃